crates/spirv-std/shared/src/image_params.rs

```diff
@@ -126,7 +123,7 @@ pub enum Sampled {
     No = 2,
 }
 
-#[cfg(any(not(target_arch = "spirv"), target_feature = "Int8"))]
+#[cfg(not(target_arch = "spirv"))]
```

Comment:
I don't think this needs a `cfg` at all, like the others

Reply:
No, it needs a `cfg`.

Reply:
Oh yeah, sorry

My eyes glazed over whilst reading it
I see now that the cfg was saying that it's only GPU compatible.

Reply:
> but now is not usable on GPU at all.

Correct, `Option<bool>` breaks with this PR, because it expects `bool` to be an `u8` - even if we forcefully disable the niche, it still turns the `bool` into `u8` as part of a scalar pair (the same `(is_some: bool, MaybeUninit<T>)`-like scalar pair representation of `Option<T>` which has allowed `for i in 0..n` to work, so we can't just mess with it in general, yet).

Reply:
Hm - with the bitcast smarts I introduced in #819 this turned out to be rather easy to amend -- will probably add that to the PR, too.